pacup

```diff
@@ -205,28 +193,40 @@ sub repology_get_newestver ( $response, $filters, $oldver, $action ) {
     my @filtered;
     my %version_count;
 
+    # List of repositories not to be used for version detection.
+    # As can be inferred, we don't like Windows versions.
+    my @BANNED_REPOS = qw(
+        appget
+        baulk
+        chocolatey
+        cygwin
+        just-install
+        scoop
+        winget
+        macports
+    );
+
     foreach my $entry (@$decoded) {
         next if grep /^\Q$entry->{'repo'}\E$/, @BANNED_REPOS;
         my $is_match
             = all { exists $entry->{$_} && $entry->{$_} eq $filters->{$_} }
             keys %$filters;
         next unless $is_match;
-        next if ( $entry->{'version'} eq 'HEAD' );
+        next if ($entry->{'version'} =~ /^(HEAD|main|master|scm|git|vcs|stable|9999)$/
+            || $entry->{'status'} eq 'rolling' && $filters->{'status'} ne 'rolling');
```

Comment:
```suggestion
        next if $entry->{'version'} =~ /^(HEAD|main|master|scm|git|vcs|stable|9999)$/
            || ( $entry->{'status'} eq 'rolling' && $filters->{'status'} ne 'rolling' );
```

Reply:
too slow

Reply:
No, my suggestion moves the parens

Reply:
wait why

Reply:
To make the precedence clear, because || and && have the same precedence. Even if it works here, the intention is not apparent